---
title: "Section 4"
author: "Syaliny"
date: "11/14/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

#Data was processed using QIIME and taxonomy was assigned using the SILVA database.For prokaryotes data was rarified at 10000 sequencing depth which removed 10 samples that had lower depths than this and for fungi data was rarified at 7000. Before input into R mapping file was modified so as to remove any special characters using the gsub function and metadata information (Treatment, Day) were added to the mapping file

```{r global_options, echo=FALSE, results='hide'}
library(knitr)
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.align='center', results='hide', echo=FALSE, warning=FALSE, message=FALSE, cache=FALSE)
```

```{r Load libraries and data}

library(phyloseq)  
library(ggplot2)
library(microbiome)
library(dplyr)
library(plyr)
library(scales)
library("MultNonParam")
library('PMCMR')
library(Rmisc)
library(RColorBrewer)
#library(dplyr)
#library(scales)
#library(reshape)
#library(reshape2)
#library(RColorBrewer)
#library(grid)

```

#Importing biom file and setting working directory for prokaryote and fungi
```{r importing data}

setwd("/Volumes/S_ GANASAMU/Paper 3/Section 4")

#Prokaryotes
# Now import the .biom-formatted otu_table-tax_table file.
Bact_biom_otu_tax <- import_biom("10000_merged_otu_table_json.biom")

# Import sample data
Bact_bmsd <- import_qiime_sample_data("Rex_antibiotic_16S_Sainur.txt")

#Now do the same for fungal data
# Now import the .biom-formatted otu_table-tax_table file.
Fungi_biom_otu_tax <- import_biom("merged_otu_table_7000.json")

# Import sample data
Fungi_bmsd <- import_qiime_sample_data("18S_2org2017edited_map.txt")
```

#Merging otu table with metadata and creating a phyloseq object
```{r merge_phyloseq_format}

#Merge into phyloseq format for prokaryote
Bacteria_phyloseq <- merge_phyloseq(Bact_biom_otu_tax, Bact_bmsd)

#check your new merged file and all your datasets should be accounted for
Bacteria_phyloseq

#Merge into phyloseq format for fungi
Fungi_phyloseq <- merge_phyloseq(Fungi_biom_otu_tax, Fungi_bmsd)

#check your new merged file and all your datasets should be accounted for
Fungi_phyloseq

```

#Attach OTU ID
```{r Attached OTU ID}

#Prokaryotes
tax_table(Bacteria_phyloseq) <- cbind(tax_table(Bacteria_phyloseq), OTU=taxa_names(Bacteria_phyloseq))

#Fungi
tax_table(Fungi_phyloseq) <- cbind(tax_table(Fungi_phyloseq), OTU=taxa_names(Fungi_phyloseq))
```

#Renaming ranks
```{r Rename Ranks}

#Prokaryotes
colnames(tax_table(Bacteria_phyloseq)) = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species", "OTU")
#This step deletes extra characters that indicate the taxonomic level. If you wish to plot class, family, genus, etc, you will need to repeat this step replacing replacing the number with the level you wish to remove. 
tax_table(Bacteria_phyloseq) =gsub("D_0__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_1__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_2__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_3__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_4__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_5__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_6__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_7__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_8__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_9__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_10__", "", tax_table(Bacteria_phyloseq))
tax_table(Bacteria_phyloseq) =gsub("D_11__", "", tax_table(Bacteria_phyloseq))

#Fungi
colnames(tax_table(Fungi_phyloseq)) = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species", "OTU")
#This step deletes extra characters that indicate the taxonomic level. If you wish to plot class, family, genus, etc, you will need to repeat this step replacing replacing the number with the level you wish to remove.
tax_table(Fungi_phyloseq) =gsub("D_0__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_1__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_2__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_3__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_4__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_5__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_6__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_7__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_8__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_9__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_10__", "", tax_table(Fungi_phyloseq))
tax_table(Fungi_phyloseq) =gsub("D_11__", "", tax_table(Fungi_phyloseq))
```

#Prune taxa for fungi keep only those that are fungi or fungal like because kingdom eukaryota includes other things as well
```{r prune taxa for fungi only}

#Subset to only keep fungi and those that are fungal like
Fungi_phyloseq = subset_taxa(Fungi_phyloseq , Order == "Fungi" |Order == "Hyphochytriales" | Order == "Labyrinthulomycetes" | Order == "Peronosporomycetes")

```

#Avergae results for multiple rarefractions
```{r Create average result for multiple rarefaction by transforming data using (divide by 10)}

#Prokaryotes
Bacteria_phyloseq = transform_sample_counts(Bacteria_phyloseq, function(x) x/10)
sample_sums(Bacteria_phyloseq)

#Fungi
Fungi_phyloseq = transform_sample_counts(Fungi_phyloseq, function(x) x/10)
sample_sums(Fungi_phyloseq)

```

#Round off and confirm count number
```{r Round and confirm count number}

#Prokaryotes
Bacteria_phyloseq = transform_sample_counts(Bacteria_phyloseq, round)
sample_sums(Bacteria_phyloseq)
Bacteria_phyloseq = prune_samples(sample_sums(Bacteria_phyloseq)>=1, Bacteria_phyloseq)
sample_sums(Bacteria_phyloseq)

#Fungi
Fungi_phyloseq = transform_sample_counts(Fungi_phyloseq, round)
sample_sums(Fungi_phyloseq)
Fungi_phyloseq = prune_samples(sample_sums(Fungi_phyloseq)>=1, Fungi_phyloseq)
sample_sums(Fungi_phyloseq)

```

#Identify if there are taxa which don't have a count 
```{r identify taxa with only zeros}

#Prokaryotes
sum(taxa_sums(Bacteria_phyloseq) > 0)
any(taxa_sums(Bacteria_phyloseq)== 0)
sum(taxa_sums(Bacteria_phyloseq) == 0)
any(taxa_sums(Bacteria_phyloseq) > 1)
sum(taxa_sums(Bacteria_phyloseq) > 1)
any(taxa_sums(Bacteria_phyloseq) < 1)
sum(taxa_sums(Bacteria_phyloseq) < 1)

#Fungi
sum(taxa_sums(Fungi_phyloseq) > 0)
any(taxa_sums(Fungi_phyloseq)== 0)
sum(taxa_sums(Fungi_phyloseq) == 0)
any(taxa_sums(Fungi_phyloseq) > 1)
sum(taxa_sums(Fungi_phyloseq) > 1)
any(taxa_sums(Fungi_phyloseq) < 1)
sum(taxa_sums(Fungi_phyloseq) < 1)
```

#Create new file with taxa that have count data
```{r  Save original file and create new file with only present (no zeroes) taxa}

#Create new file with only present (no zeroes) taxa
#Prokaryotes
Bacteria_phyloseq = prune_taxa(taxa_sums(Bacteria_phyloseq) > 1, Bacteria_phyloseq)
any(sample_sums(Bacteria_phyloseq) == 0)
any(sample_sums(Bacteria_phyloseq) > 0)
sum(taxa_sums(Bacteria_phyloseq) > 0)
any(sample_sums(Bacteria_phyloseq) < 1)
sum(taxa_sums(Bacteria_phyloseq) < 1)

#Fungi
Fungi_phyloseq = prune_taxa(taxa_sums(Fungi_phyloseq) > 1, Fungi_phyloseq)
any(sample_sums(Fungi_phyloseq) == 0)
any(sample_sums(Fungi_phyloseq) > 0)
sum(taxa_sums(Fungi_phyloseq) > 0)
any(sample_sums(Fungi_phyloseq) < 1)
sum(taxa_sums(Fungi_phyloseq) < 1)
```

#Compare sequences per sample or OTU
```{r Compare sequences per sample or OTU}

#Prokaryotes
Bactreadsumsdf = data.frame(nreads = sort(taxa_sums(Bacteria_phyloseq),TRUE), sorted = 1:ntaxa(Bacteria_phyloseq), type = "OTU")
Bactreadsumsdf = rbind(Bactreadsumsdf,data.frame(nreads = sort(sample_sums(Bacteria_phyloseq),TRUE),sorted = 1:nsamples(Bacteria_phyloseq), type = "Samples"))

title = "Total number of reads"

Bact_seq_persample = ggplot(Bactreadsumsdf, aes(x = sorted, y = nreads)) +
  geom_bar(stat = "identity")

Bact_seq_persample + ggtitle(title) +
  scale_y_log10() +
  facet_wrap(~type, 1, scales = "free")

#Fungi
Fungireadsumsdf = data.frame(nreads = sort(taxa_sums(Fungi_phyloseq),TRUE), sorted = 1:ntaxa(Fungi_phyloseq), type = "OTU")
Fungireadsumsdf = rbind(Fungireadsumsdf,data.frame(nreads = sort(sample_sums(Fungi_phyloseq),TRUE),sorted = 1:nsamples(Fungi_phyloseq), type = "Samples"))

title = "Total number of reads"

Fungi_seq_persample = ggplot(Fungireadsumsdf, aes(x = sorted, y = nreads)) +
  geom_bar(stat = "identity")

Fungi_seq_persample + ggtitle(title) +
  scale_y_log10() +
  facet_wrap(~type, 1, scales = "free")
```

#Figure 5
#Based on Figure 2, the most sensitive time point where a lot of changes were happening (high bray disimilarrity value) was identified as Day 9 for prokaryotes
#Exact test Prokaryotes to identify organisms that were significantly changing on Day 9
#Exact test separate by time point
#Now separate time point by treatment, time points are separated by positive control (urea) against inhibitor treatments (urea+inhibitors) ,this is done to identify organisms that were significantly responding to urea and inhibitor treatment 
```{r Exact test time point 1}

#Rename Antibacterial_Antibacterial to Antibacterial+Antifungal
levels(sample_data(Bacteria_phyloseq)$Treatment)[levels(sample_data(Bacteria_phyloseq)$Treatment)=="Antibacterial_Antifungal"]="Antibacterial+Antifungal"

#Check if the renaming was done
levels(sample_data(Bacteria_phyloseq)$Treatment)

#Subset by time point from phyloseq
#Time point 1 Day 9
Bact_T1<-subset_samples(Bacteria_phyloseq,Day%in%c("9"))
sample_data(Bact_T1)$Day

#Subset by treatment
Bact_T1_PC_AB=subset_samples(Bact_T1,Treatment%in% c("Positive Control", "Antibacterial"))
unique(sample_data(Bact_T1_PC_AB)$Treatment) #To check if subset was done correctly

Bact_T1_PC_AF=subset_samples(Bact_T1,Treatment%in% c("Positive Control", "Antifungal"))
unique(sample_data(Bact_T1_PC_AF)$Treatment) #To check if subset was done correctly

Bact_T1_PC_AB_AF=subset_samples(Bact_T1,Treatment%in% c("Positive Control", "Antibacterial+Antifungal"))
unique(sample_data(Bact_T1_PC_AB_AF)$Treatment) #To check if subset was done correctly
```

#Time point 1 Day 9
#ET Time point 1 PC vs AB
```{r Exact time point 1 PC vs AB}
#check open packages
(.packages())
##Close all
detachAllPackages <- function() {
  
  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")
  
  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]
  
  package.list <- setdiff(package.list,basic.packages)
  
  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE)
  
}

detachAllPackages()
library(phyloseq)  
library(ggplot2)
library(microbiome)
library(dplyr)
library(plyr)
library(scales)
library("MultNonParam")
library('PMCMR')

phyloseq_to_edgeR = function(physeq, group, method = "RLE", ...) {
  require("edgeR")
  require("limma")
  require("phyloseq")
  # Enforce orientation.
  if (!taxa_are_rows(physeq)) {
    physeq <- t(physeq)
  }
  x = as(otu_table(physeq), "matrix")
  # Add one to protect against overflow, log(0) issues.
  x = x + 1
  # Check `group` argument
  if (identical(all.equal(length(group), 1), TRUE) & nsamples(physeq) > 1) {
    # Assume that group was a sample variable name (must be categorical)
    group = get_variable(physeq, group)
  }
  # Define gene annotations (`genes`) as tax_table
  taxonomy = tax_table(physeq, errorIfNULL=FALSE)
  if( !is.null(taxonomy) ){
    taxonomy = data.frame(as(taxonomy, "matrix"))
  } 
  # Now turn into a DGEList
  y = DGEList(counts = x, group = group, genes = taxonomy, remove.zeros = TRUE, 
              ...)
  # Calculate the normalization factors
  z = calcNormFactors(y, method = method)
  # Check for division by zero inside `calcNormFactors`
  if (!all(is.finite(z$samples$norm.factors))) {
    stop("Something wrong with edgeR::calcNormFactors on this data,\n         non-finite $norm.factors, consider changing `method` argument")
  }
  # Estimate dispersions
  return(estimateTagwiseDisp(estimateCommonDisp(z)))
}


dge1 = phyloseq_to_edgeR(Bact_T1_PC_AB, group = "Treatment")

# Perform binary test
et1 = exactTest(dge1)
# Extract values from test results
tt1 = topTags(et1, n = nrow(dge1$table), adjust.method = "BH", sort.by = "PValue")
res1 = tt1@.Data[[1]]
alpha = 0.05
B_sigtab_T1_PC_AB = res1[(res1$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                  
B_sigtab_2fold_T1_PC_AB <- subset(B_sigtab_T1_PC_AB, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(B_sigtab_2fold_T1_PC_AB)

#This ensure that the df starts with the OTU column otherwise there is an error when trying to subset phyloseq 
B_sigtab_2fold_T1_PC_AB <- data.frame(OTU = rownames(B_sigtab_2fold_T1_PC_AB), B_sigtab_2fold_T1_PC_AB)

```


#ET Time point 1 PC vs AF
```{r Exact time point 1 PC vs AF}

dge2 = phyloseq_to_edgeR(Bact_T1_PC_AF, group = "Treatment")

# Perform binary test
et2 = exactTest(dge2)
# Extract values from test results
tt2 = topTags(et2, n = nrow(dge2$table), adjust.method = "BH", sort.by = "PValue")
res2 = tt2@.Data[[1]]
alpha = 0.05
B_sigtab_T1_PC_AF = res2[(res2$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                              
B_sigtab_2fold_T1_PC_AF <- subset(B_sigtab_T1_PC_AF, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(B_sigtab_2fold_T1_PC_AF)

#This ensure that the df starts with the OTU column otherwise there is an error when trying to subset phyloseq 
B_sigtab_2fold_T1_PC_AF <- data.frame(OTU = rownames(B_sigtab_2fold_T1_PC_AF), B_sigtab_2fold_T1_PC_AF)

```


#ET Time point 1 PC vs AB_AF 
```{r Exact time point 1 PC vs AB_AF}

dge3 = phyloseq_to_edgeR(Bact_T1_PC_AB_AF, group = "Treatment")

# Perform binary test
et3 = exactTest(dge3)
# Extract values from test results
tt3 = topTags(et3, n = nrow(dge3$table), adjust.method = "BH", sort.by = "PValue")
res3 = tt3@.Data[[1]]
alpha = 0.05
B_sigtab_T1_PC_AB_AF = res3[(res3$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                 
B_sigtab_2fold_T1_PC_AB_AF <- subset(B_sigtab_T1_PC_AB_AF, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(B_sigtab_2fold_T1_PC_AB_AF)

#This ensure that the df starts with the OTU column otherwise there is an error when trying to subset phyloseq 
B_sigtab_2fold_T1_PC_AB_AF <- data.frame(OTU = rownames(B_sigtab_2fold_T1_PC_AB_AF), B_sigtab_2fold_T1_PC_AB_AF)


```


#Large df for ET at time point 1
#Make one list of which the organisms have been shown to change from the Exact test.
#Combine exact test at the genus level 
```{r Make one dataframe for sig OTU's at Genus level }

#make one data frame of ET according to treatment then subset phyloseq to only keep the significant ones at that particular time point
B_FDRcorrected_T1=c(as.character(B_sigtab_2fold_T1_PC_AB[,1]),as.character(B_sigtab_2fold_T1_PC_AF[,1]), as.character(B_sigtab_2fold_T1_PC_AB_AF[,1]))

#Subset the taxa by the OTUs that were shown to change significantly) 
B_Subset_sig_OTU_T1_genus <-subset_taxa(Bacteria_phyloseq, OTU %in% B_FDRcorrected_T1) 

#Write csv to edit names in excel
write.csv(tax_table(B_Subset_sig_OTU_T1_genus),file="B_sigOTU_T1.csv")

#import csv with changed taxa names, remove first col of OTU names from B_sigOTU_T1.csv and save edited version after changing taxa names
B_Subset_sig_OTU_T1_genus_edited=read.csv("/Volumes/S_ GANASAMU/Paper 3/Section 4/B_sigOTU_T1_edited.csv")

#Edit row names so that it is in the same format as taxonomy table in the phyloseq obj otherwise can't merge
row.names(B_Subset_sig_OTU_T1_genus_edited)<-B_Subset_sig_OTU_T1_genus_edited$OTU

#create matrix from edited list
BT1=as.matrix(B_Subset_sig_OTU_T1_genus_edited)
taxBT1=tax_table(BT1) #create taxonomy table with edited genus names
tax_table(B_Subset_sig_OTU_T1_genus)=taxBT1 #change taxonomy table in original phyloseq with edited taxonomy table


#Glom at genus level
B_Subset_sig_OTU_T1_genus_df <- tax_glom(B_Subset_sig_OTU_T1_genus, taxrank = 'Genus') %>%#Merge the species at the genus level
psmelt() #Melt it into a dataframe  #Error no taxonomy table

```

#Manually sort data by response towards urea and inhibitors by adding these two columns to data frame in excel and then export back into R
```{r summary of significant OTU}

#Calculate summary 
B_All_sig_OTU_Gen_T1_df <- summarySE(B_Subset_sig_OTU_T1_genus_df, measurevar = "Abundance", groupvars=c("Treatment", "Genus","Phylum"))


B_All_sig_OTU_Gen_T1_df<-dplyr::arrange(B_All_sig_OTU_Gen_T1_df, Genus, Abundance) #Arrange the rows by listed variables

#write csv for B_All_sig_OTU_Gen_T1_df and manually add columns for response towards urea and inhibitors
write.csv(B_All_sig_OTU_Gen_T1_df,file="B_All_sig_OTU_Gen_T1_df.csv")

#read file response columns added to it,edit Proteobacteria col in B_All_sig_OTU_Gen_T1_df to Alpha,Beta,Delta and Gamma
B_T1_growth_response=read.csv("/Volumes/S_ GANASAMU/Paper 3/Section 4/B_All_sig_OTU_Gen_T1_df_edited.csv")

```


#Scale colour manual prokaryotes 
```{r colour manual}

#Choosing colour palettes
display.brewer.all(colorblindFriendly = TRUE)
display.brewer.pal(n = 9, name = 'YlOrRd')
brewer.pal(n = 9, name = "YlOrRd")
display.brewer.pal(n = 8, name = 'Dark2')
brewer.pal(n = 8, name = "Dark2")
brewer.pal(n = 9, name = "YlGnBu")
display.brewer.pal(n = 9, name = 'YlGnBu')

Growthrescolourlist <- c(Alphaproteobacteria="#800026",Actinobacteria="#BD0026",Bacteroidetes="#FC4E2A",Betaproteobacteria="#FD8D3C",Deltaproteobacteria="#FED976",Firmicutes="#E6AB02",Gemmatimonadetes="#A6761D",Acidobacteria="#1D91C0",Gammaproteobacteria="#253494",Verrucomicrobia="#081D58",Planctomycetes="#7FCDBB")

```


#Subset by response towards urea and inhibitors
```{r}

B_T1_growth_response<-dplyr::arrange(B_T1_growth_response, Response_to_urea,Response_to_Inhibitors) #Arrange the rows by listed variables

B_T1_growth_response$Treatment<-factor(B_T1_growth_response$Treatment, levels=c("Negative Control","Positive Control", "Antibacterial", "Antifungal","Antibacterial+Antifungal"))

#Subset df by Reponse towards urea and response towards inhibitors
BT1_Positive_Stimulated=subset(B_T1_growth_response,Response_to_urea=="Positive" & Response_to_Inhibitors=="Stimulated")

BT1_Positive_Inhibited=subset(B_T1_growth_response,Response_to_urea=="Positive" & Response_to_Inhibitors=="Inhibited")

BT1_Negative_Stimulated=subset(B_T1_growth_response,Response_to_urea=="Negative" & Response_to_Inhibitors=="Stimulated" )

BT1_Negative_Inhibited=subset(B_T1_growth_response,Response_to_urea=="Negative" & Response_to_Inhibitors=="Inhibited")

```

#Figure 5
#Plot Positive_Stimulated
```{r}
#export and save pdf plots as landscape 12 x 20

#Plot
pd=position_dodge(0.1)
BT1_Positive_Stimulated_plot=ggplot(BT1_Positive_Stimulated, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Phylum))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1,size=20),strip.text = element_text(size=20),axis.text.y = element_text(size=20), axis.title=element_text(size=25,face="bold"),legend.text=element_text(size=25),legend.title=element_text(size=25,face="bold"))+
 scale_fill_manual(values=Growthrescolourlist)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=4,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x tick mark labels
BT1_Positive_Stimulated_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

```

#Figure 5
#Plot Positive_Inhibited
```{r}

#export and save pdf plots as landscape 12 x 20

#Plot
BT1_Positive_Inhibited_plot=ggplot(BT1_Positive_Inhibited, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Phylum))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
   theme(axis.text.x = element_text(angle = 45, hjust = 1,size=20),strip.text = element_text(size=20),axis.text.y = element_text(size=20), axis.title=element_text(size=25,face="bold"),legend.text=element_text(size=25),legend.title=element_text(size=25,face="bold"))+
 scale_fill_manual(values=Growthrescolourlist)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=4,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x tick mark labels
BT1_Positive_Inhibited_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

```

#Fugure 5
#Plot Negative_Stimulated
```{r}

#export and save pdf plots as landscape 10 x 20

#Plot
BT1_Negative_Stimulated_plot=ggplot(BT1_Negative_Stimulated, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Phylum))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1,size=20),strip.text = element_text(size=20),axis.text.y = element_text(size=20), axis.title=element_text(size=25,face="bold"),legend.text=element_text(size=25),legend.title=element_text(size=25,face="bold"))+
 scale_fill_manual(values=Growthrescolourlist)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=4,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x tick mark labels
BT1_Negative_Stimulated_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))
```


#Figure 6
#Based on Figure 2, the most sensitive time point where a lot of changes were happening (high bray disimilarrity value) was identified as Day 9,Day 33 and Day 42 for fungi
#Exact test fungi to identify organisms that were significantly changing on Day 9, Day 33 and Day 42
#Exact test separate by time point
#Now separate time point by treatment, time points are separated by positive control (urea) against inhibitor treatments (urea+inhibitors) ,this is done to identify organisms that were significantly responding to urea and inhibitor treatment 
```{r time point 1}

#Rename Antibacterial_Antibacterial to Antibacterial+Antifungal
levels(sample_data(Fungi_phyloseq)$Treatment)[levels(sample_data(Fungi_phyloseq)$Treatment)=="Antibacterial_Antifungal"]="Antibacterial+Antifungal"

#Check if the renaming was done
levels(sample_data(Fungi_phyloseq)$Treatment)

#Subset by time point from phyloseq
#Time point Day 9
Fungi_T1<-subset_samples(Fungi_phyloseq,Day%in%c("9"))
sample_data(Fungi_T1)$Day

Fungi_T1_PC_AB=subset_samples(Fungi_T1,Treatment%in% c("Positive Control", "Antibacterial"))
unique(sample_data(Fungi_T1_PC_AB)$Treatment) #To check if subset was done correctly

Fungi_T1_PC_AF=subset_samples(Fungi_T1,Treatment%in% c("Positive Control", "Antifungal"))
unique(sample_data(Fungi_T1_PC_AF)$Treatment) #To check if subset was done correctly

Fungi_T1_PC_AB_AF=subset_samples(Fungi_T1,Treatment%in% c("Positive Control", "Antibacterial+Antifungal"))
unique(sample_data(Fungi_T1_PC_AB_AF)$Treatment) #To check if subset was done correctly

```

#Time point 1 Day 9
#ET Time point 1 PC vs AB
```{r Exact time point 1 PC vs AB}
#check open packages
(.packages())
##Close all
detachAllPackages <- function() {
  
  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")
  
  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]
  
  package.list <- setdiff(package.list,basic.packages)
  
  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE)
  
}

detachAllPackages()


library(phyloseq)  
library(ggplot2)
library(microbiome)
library(dplyr)
library(plyr)
library(scales)
library("MultNonParam")
library('PMCMR')


#Convert phyloseq OTU count data into DGEList for edgeR package
##http://joey711.github.io/phyloseq-extensions/edgeR.html

#to install edgeR
#source("https://bioconductor.org/biocLite.R")
#biocLite("edgeR")

phyloseq_to_edgeR = function(physeq, group, method = "RLE", ...) {
  require("edgeR")
  require("limma")
  require("phyloseq")
  # Enforce orientation.
  if (!taxa_are_rows(physeq)) {
    physeq <- t(physeq)
  }
  x = as(otu_table(physeq), "matrix")
  # Add one to protect against overflow, log(0) issues.
  x = x + 1
  # Check `group` argument
  if (identical(all.equal(length(group), 1), TRUE) & nsamples(physeq) > 1) {
    # Assume that group was a sample variable name (must be categorical)
    group = get_variable(physeq, group)
  }
  # Define gene annotations (`genes`) as tax_table
  taxonomy = tax_table(physeq, errorIfNULL=FALSE)
  if( !is.null(taxonomy) ){
    taxonomy = data.frame(as(taxonomy, "matrix"))
  } 
  # Now turn into a DGEList
  y = DGEList(counts = x, group = group, genes = taxonomy, remove.zeros = TRUE, 
              ...)
  # Calculate the normalization factors
  z = calcNormFactors(y, method = method)
  # Check for division by zero inside `calcNormFactors`
  if (!all(is.finite(z$samples$norm.factors))) {
    stop("Something wrong with edgeR::calcNormFactors on this data,\n         non-finite $norm.factors, consider changing `method` argument")
  }
  # Estimate dispersions
  return(estimateTagwiseDisp(estimateCommonDisp(z)))
}


dge4 = phyloseq_to_edgeR(Fungi_T1_PC_AB, group = "Treatment")

# Perform binary test
et4 = exactTest(dge4)
# Extract values from test results
tt4 = topTags(et4, n = nrow(dge4$table), adjust.method = "BH", sort.by = "PValue")
res4 = tt4@.Data[[1]]
alpha = 0.05
F_sigtab_T1_PC_AB = res4[(res4$FDR <= alpha), ]


#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                                        
F_sigtab_2fold_T1_PC_AB <- subset(F_sigtab_T1_PC_AB, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T1_PC_AB)

#
F_sigtab_2fold_T1_PC_AB <- data.frame(OTU = rownames(F_sigtab_2fold_T1_PC_AB), F_sigtab_2fold_T1_PC_AB)

```


#ET Time point 1 PC vs AF (F sigtab2f=0)
```{r Exact time point 1 PC vs AF}

dge5 = phyloseq_to_edgeR(Fungi_T1_PC_AF, group = "Treatment")

# Perform binary test
et5 = exactTest(dge5)
# Extract values from test results
tt5 = topTags(et5, n = nrow(dge5$table), adjust.method = "BH", sort.by = "PValue")
res5 = tt5@.Data[[1]]
alpha = 0.05
F_sigtab_T1_PC_AF = res5[(res5$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                                         
F_sigtab_2fold_T1_PC_AF <- subset(F_sigtab_T1_PC_AF, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T1_PC_AF)

#
#F_sigtab_2fold_T1_PC_AF <- data.frame(OTU = rownames(F_sigtab_2fold_T1_PC_AF), F_sigtab_2fold_T1_PC_AF)


```


#ET Time point 1 PC vs AB_AF (F sigtab=0)
```{r Exact time point 1 PC vs AB_AF}

dge6 = phyloseq_to_edgeR(Fungi_T1_PC_AB_AF, group = "Treatment")

# Perform binary test
et6 = exactTest(dge6)
# Extract values from test results
tt6 = topTags(et6, n = nrow(dge6$table), adjust.method = "BH", sort.by = "PValue")
res6 = tt6@.Data[[1]]
alpha = 0.05
F_sigtab_T1_PC_AB_AF = res6[(res6$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                                                 
F_sigtab_2fold_T1_PC_AB_AF <- subset(F_sigtab_T1_PC_AB_AF, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T1_PC_AB_AF)

#
#F_sigtab_2fold_T1_PC_AB_AF <- data.frame(OTU = rownames(F_sigtab_2fold_T1_PC_AB_AF), F_sigtab_2fold_T1_PC_AB_AF)

```


#Large df for ET at time point 1
#Make one list of which the organisms have been shown to change from the Exact test.
#Combine exact test at the genus level 
```{r Make one dataframe for sig OTU's at Genus level }

F_All_sig_OTU_T1_genus <- c(as.character(F_sigtab_2fold_T1_PC_AB[,1])) #Extract the OTU table that was shown to be significant

F_Subset_sig_OTU_T1_genus <- subset_taxa(Fungi_phyloseq, OTU %in% F_All_sig_OTU_T1_genus) #Subset the taxa by the OTUs that were shown to change significantly

#Write csv to edit names in excel
write.csv(tax_table(F_Subset_sig_OTU_T1_genus),file="F_sigOTU_T1.csv")

#import csv with changed taxa names
F_Subset_sig_OTU_T1_genus_edited=read.csv("/Volumes/S_ GANASAMU/Paper 3/Section 4/F_sigOTU_T1_edited.csv")

#Edit row names so that it is in the same format as taxonomy table in the phyloseq obj otherwise can't merge
row.names(F_Subset_sig_OTU_T1_genus_edited)<-F_Subset_sig_OTU_T1_genus_edited$OTU

#create matrix from edited list
FT1=as.matrix(F_Subset_sig_OTU_T1_genus_edited)
taxFT1=tax_table(FT1) #create taxonomy table with edited genus names
tax_table(F_Subset_sig_OTU_T1_genus)=taxFT1 #change taxonomy table in original phyloseq with edited taxonomy table


#Glom at genus level
F_Subset_sig_OTU_T1_genus_df <- tax_glom(F_Subset_sig_OTU_T1_genus, taxrank = 'Genus') %>%#Merge the species at the genus level
psmelt() #Melt it into a dataframe  
  

```

#Manually sort data by response towards urea and inhibitors by adding these two columns to data frame in excel and then export back into R
```{r summary of significant OTU}

#Calculate summary 
F_All_sig_OTU_Gen_T1_df <- summarySE(F_Subset_sig_OTU_T1_genus_df, measurevar = "Abundance", groupvars=c("Treatment", "Genus","Phylum"))


F_All_sig_OTU_Gen_T1_df<-dplyr::arrange(F_All_sig_OTU_Gen_T1_df, Genus, Abundance) #Arrange the rows by listed variables

#write csv for B_All_sig_OTU_Gen_T1_df and manually add columns for growth strategy and response towards antimicrobials 
write.csv(F_All_sig_OTU_Gen_T1_df,file="F_All_sig_OTU_Gen_T1_df.csv")

#read file with response towards urea and inhibitor columns added to it
F_T1_growth_response=read.csv("/Volumes/S_ GANASAMU/Paper 3/Section 4/F_All_sig_OTU_Gen_T1_df_edited.csv")


```


#Scale colour manual fungi
```{r}

FungiGrowthrescolourlistT1 <- c(Ascobolaceae="#084081",Brachyconidiellopsis="#800026",Helotiales="#BD0026",Mortierellales="#0868AC",Nectria="#FC4E2A")

FungiGrowthrescolourlistT5 <- c(Agaricomycetes="#E31A1C",Tremellales="#2B8CBE")

FungiGrowthrescolourlistT6 <- c(LKM11="#662506",Pleosporales="#993404",Tremellales="#CC4C02",Tapinella="#4EB3D3",Tremella="#A8DDB5",Helotiales="#EC7014")

```


#Subset by response towards to urea and inhibitors
```{r}

F_T1_growth_response<-dplyr::arrange(F_T1_growth_response, Response_to_urea,Response_to_Inhibitors) #Arrange the rows by listed variables


F_T1_growth_response$Treatment<-factor(F_T1_growth_response$Treatment, levels=c("Negative Control","Positive Control", "Antibacterial", "Antifungal","Antibacterial+Antifungal"))

#Subset by response towards urea and inhibitors
FT1_Positive_Stimulated=subset(F_T1_growth_response,Response_to_urea=="Positive" & Response_to_Inhibitors=="Stimulated")

FT1_Negative_Stimulated=subset(F_T1_growth_response,Response_to_urea=="Negative" & Response_to_Inhibitors=="Stimulated")

```

#Figure 6
#T1 Plot Positive_Stimulated
```{r}
#save plot using ggsave 

#Plot
FT1_Positive_Stimulated_plot=ggplot(FT1_Positive_Stimulated, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Genus))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
 theme(axis.text.x = element_text(angle = 45, hjust = 1,size=15),strip.text = element_text(size=15),axis.text.y = element_text(size=15), axis.title=element_text(size=20,face="bold"),legend.text=element_text(size=20),legend.title=element_text(size=20,face="bold"))+
 scale_fill_manual(values=FungiGrowthrescolourlistT1)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=3,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x axis tick mark label
FT1_Positive_Stimulated_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

ggsave("/Volumes/S_ GANASAMU/Paper 3/Section 4/1.pdf",width=12,height=3)
```

#Figure 6
#T1 Plot Negative_Stimulated
```{r}

#save plot using ggsave

#Plot
FT1_Negative_Stimulated_plot=ggplot(FT1_Negative_Stimulated, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Genus))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
 theme(axis.text.x = element_text(angle = 45, hjust = 1,size=15),strip.text = element_text(size=15),axis.text.y = element_text(size=15), axis.title=element_text(size=20,face="bold"),legend.text=element_text(size=20),legend.title=element_text(size=20,face="bold"))+
 scale_fill_manual(values=FungiGrowthrescolourlistT1)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=2,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x axis tick mark label
FT1_Negative_Stimulated_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

ggsave("/Volumes/S_ GANASAMU/Paper 3/Section 4/2.pdf",width=9,height=3)
```



#Exact test separate by time point and then by treatment 
```{r Exact test time point 5 }

#Subset by time point from phyloseq
#Time point 5 Day 33
Fungi_T5<-subset_samples(Fungi_phyloseq,Day%in%c("33"))
sample_data(Fungi_T5)$Day

#Subset by treatment
Fungi_T5_PC_AB=subset_samples(Fungi_T5,Treatment%in% c("Positive Control", "Antibacterial"))
unique(sample_data(Fungi_T5_PC_AB)$Treatment) #To check if subset was done correctly

Fungi_T5_PC_AF=subset_samples(Fungi_T5,Treatment%in% c("Positive Control", "Antifungal"))
unique(sample_data(Fungi_T5_PC_AF)$Treatment) #To check if subset was done correctly

Fungi_T5_PC_AB_AF=subset_samples(Fungi_T5,Treatment%in% c("Positive Control", "Antibacterial+Antifungal"))
unique(sample_data(Fungi_T5_PC_AB_AF)$Treatment) #To check if subset was done correctly
```


#Time point 5 Day 33
#ET Time point 5 PC vs AB (Fungi) 
```{r Exact time point 5 PC vs AB}
#check open packages
(.packages())
##Close all
detachAllPackages <- function() {
  
  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")
  
  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]
  
  package.list <- setdiff(package.list,basic.packages)
  
  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE)
  
}

detachAllPackages()
library(phyloseq)  
library(ggplot2)
library(microbiome)
library(dplyr)
library(plyr)
library(scales)
library("MultNonParam")
library('PMCMR')

phyloseq_to_edgeR = function(physeq, group, method = "RLE", ...) {
  require("edgeR")
  require("limma")
  require("phyloseq")
  # Enforce orientation.
  if (!taxa_are_rows(physeq)) {
    physeq <- t(physeq)
  }
  x = as(otu_table(physeq), "matrix")
  # Add one to protect against overflow, log(0) issues.
  x = x + 1
  # Check `group` argument
  if (identical(all.equal(length(group), 1), TRUE) & nsamples(physeq) > 1) {
    # Assume that group was a sample variable name (must be categorical)
    group = get_variable(physeq, group)
  }
  # Define gene annotations (`genes`) as tax_table
  taxonomy = tax_table(physeq, errorIfNULL=FALSE)
  if( !is.null(taxonomy) ){
    taxonomy = data.frame(as(taxonomy, "matrix"))
  } 
  # Now turn into a DGEList
  y = DGEList(counts = x, group = group, genes = taxonomy, remove.zeros = TRUE, 
              ...)
  # Calculate the normalization factors
  z = calcNormFactors(y, method = method)
  # Check for division by zero inside `calcNormFactors`
  if (!all(is.finite(z$samples$norm.factors))) {
    stop("Something wrong with edgeR::calcNormFactors on this data,\n         non-finite $norm.factors, consider changing `method` argument")
  }
  # Estimate dispersions
  return(estimateTagwiseDisp(estimateCommonDisp(z)))
}


dge7 = phyloseq_to_edgeR(Fungi_T5_PC_AB, group = "Treatment")

# Perform binary test
et7 = exactTest(dge7)
# Extract values from test results
tt7 = topTags(et7, n = nrow(dge7$table), adjust.method = "BH", sort.by = "PValue")
res7 = tt7@.Data[[1]]
alpha = 0.05
F_sigtab_T5_PC_AB = res7[(res7$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                  
F_sigtab_2fold_T5_PC_AB <- subset(F_sigtab_T5_PC_AB, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T5_PC_AB)

#This ensure that the df starts with the OTU column otherwise there is an error when trying to subset phyloseq 
F_sigtab_2fold_T5_PC_AB <- data.frame(OTU = rownames(F_sigtab_2fold_T5_PC_AB), F_sigtab_2fold_T5_PC_AB)

```


#ET Time point 5 PC vs AF (Fungi)
```{r Exact time point 5 PC vs AF}

dge8 = phyloseq_to_edgeR(Fungi_T5_PC_AF, group = "Treatment")

# Perform binary test
et8 = exactTest(dge8)
# Extract values from test results
tt8 = topTags(et8, n = nrow(dge8$table), adjust.method = "BH", sort.by = "PValue")
res8 = tt8@.Data[[1]]
alpha = 0.05
F_sigtab_T5_PC_AF = res8[(res8$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                              
F_sigtab_2fold_T5_PC_AF <- subset(F_sigtab_T5_PC_AF, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T5_PC_AF)

#This ensure that the df starts with the OTU column otherwise there is an error when trying to subset phyloseq 
F_sigtab_2fold_T5_PC_AF <- data.frame(OTU = rownames(F_sigtab_2fold_T5_PC_AF), F_sigtab_2fold_T5_PC_AF)

```


#ET Time point 5 PC vs AB_AF (Fungi)
```{r Exact time point 5 PC vs AB_AF}

dge9 = phyloseq_to_edgeR(Fungi_T5_PC_AB_AF, group = "Treatment")

# Perform binary test
et9 = exactTest(dge9)
# Extract values from test results
tt9 = topTags(et9, n = nrow(dge9$table), adjust.method = "BH", sort.by = "PValue")
res9 = tt9@.Data[[1]]
alpha = 0.05
F_sigtab_T5_PC_AB_AF = res9[(res9$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                 
F_sigtab_2fold_T5_PC_AB_AF <- subset(F_sigtab_T5_PC_AB_AF, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T5_PC_AB_AF)

#This ensure that the df starts with the OTU column otherwise there is an error when trying to subset phyloseq 
F_sigtab_2fold_T5_PC_AB_AF <- data.frame(OTU = rownames(F_sigtab_2fold_T5_PC_AB_AF), F_sigtab_2fold_T5_PC_AB_AF)


```


#Large df for ET at time point 5
#Make one list of which the organisms have been shown to change from the Exact test.
#Combine exact test at the genus level 
```{r Make one dataframe for sig OTU's at Genus level }

#make one data frame of ET according to treatment then subset phyloseq to only keep the significant ones at that particular time point
F_FDRcorrected_T5=c(as.character(F_sigtab_2fold_T5_PC_AB[,1]),as.character(F_sigtab_2fold_T5_PC_AF[,1]), as.character(F_sigtab_2fold_T5_PC_AB_AF[,1]))

#Subset the taxa by the OTUs that were shown to change significantly) 
F_Subset_sig_OTU_T5_genus <-subset_taxa(Fungi_phyloseq, OTU %in% F_FDRcorrected_T5) 

#Write csv to edit names in excel but since all names are informative no editting needed
write.csv(tax_table(F_Subset_sig_OTU_T5_genus),file="F_sigOTU_T5.csv")

#Glom at genus level
F_Subset_sig_OTU_T5_genus_df <- tax_glom(F_Subset_sig_OTU_T5_genus, taxrank = 'Genus') %>%#Merge the species at the genus level
psmelt() #Melt it into a dataframe  #Error no taxonomy table

```


#Manually sort data by response towards urea and inhibitors by adding these two columns to data frame in excel and then export back into R
```{r summary of significant OTU}

#Calculate summary 
F_All_sig_OTU_Gen_T5_df <- summarySE(F_Subset_sig_OTU_T5_genus_df, measurevar = "Abundance", groupvars=c("Treatment", "Genus","Phylum"))


F_All_sig_OTU_Gen_T5_df<-dplyr::arrange(F_All_sig_OTU_Gen_T5_df, Genus, Abundance) #Arrange the rows by listed variables

#write csv for B_All_sig_OTU_Gen_T1_df and manually add columns for growth strategy and response towards antimicrobials 
write.csv(F_All_sig_OTU_Gen_T5_df,file="F_All_sig_OTU_Gen_T5_df.csv")  

#read file with response columns added to it,delete number col from F_All_sig_OTU_Gen_T5_df.csv
F_T5_growth_response=read.csv("/Volumes/S_ GANASAMU/Paper 3/Section 4/F_All_sig_OTU_Gen_T5_df_edited.csv")

```


#Figure 6
#T5 Plot Negative_Stimulated and Positive_Inhibited
```{r}

#save plot using ggsave

F_T5_growth_response<-dplyr::arrange(F_T5_growth_response, Response_to_urea,Response_to_Inhibitors) #Arrange the rows by listed variables


F_T5_growth_response$Treatment<-factor(F_T5_growth_response$Treatment, levels=c("Negative Control","Positive Control", "Antibacterial", "Antifungal","Antibacterial+Antifungal"))

#subset by response towards urea and inhibitors
FT5_Negative_Stimulated=subset(F_T5_growth_response,Response_to_urea=="Negative" & Response_to_Inhibitors=="Stimulated")

FT5_Positive_Inhibited=subset(F_T5_growth_response,Response_to_urea=="Positive" & Response_to_Inhibitors=="Inhibited")

#Plot
FT5_Negative_Stimulated_plot=ggplot(FT5_Negative_Stimulated, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Genus))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
 theme(axis.text.x = element_text(angle = 45, hjust = 1,size=15),strip.text = element_text(size=15),axis.text.y = element_text(size=15), axis.title=element_text(size=20,face="bold"),legend.text=element_text(size=20),legend.title=element_text(size=20,face="bold"))+
 scale_fill_manual(values=FungiGrowthrescolourlistT5)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=2,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x axis tick mark labels
FT5_Negative_Stimulated_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

ggsave("/Volumes/S_ GANASAMU/Paper 3/Section 4/3.pdf",width=6,height=3)


#Plot
FT5_Positive_Inhibited_plot=ggplot(FT5_Positive_Inhibited, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Genus))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
 theme(axis.text.x = element_text(angle = 45, hjust = 1,size=15),strip.text = element_text(size=15),axis.text.y = element_text(size=15), axis.title=element_text(size=20,face="bold"),legend.text=element_text(size=20),legend.title=element_text(size=20,face="bold"))+
 scale_fill_manual(values=FungiGrowthrescolourlistT5)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=2,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x axis tick mark labels
FT5_Positive_Inhibited_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

ggsave("/Volumes/S_ GANASAMU/Paper 3/Section 4/4.pdf",width=6,height=3)
```


#Exact test separate by time point and then by treatment 
```{r Exact test time point 6 }

#Subset by time point from phyloseq
#Time point 6 Day 42 
Fungi_T6<-subset_samples(Fungi_phyloseq,Day%in%c("42"))
sample_data(Fungi_T6)$Day

#Subset by treatment
Fungi_T6_PC_AB=subset_samples(Fungi_T6,Treatment%in% c("Positive Control", "Antibacterial"))
unique(sample_data(Fungi_T6_PC_AB)$Treatment) #To check if subset was done correctly

Fungi_T6_PC_AF=subset_samples(Fungi_T6,Treatment%in% c("Positive Control", "Antifungal"))
unique(sample_data(Fungi_T6_PC_AF)$Treatment) #To check if subset was done correctly

Fungi_T6_PC_AB_AF=subset_samples(Fungi_T6,Treatment%in% c("Positive Control", "Antibacterial+Antifungal"))
unique(sample_data(Fungi_T6_PC_AB_AF)$Treatment) #To check if subset was done correctly
```


#Time point 6 Day 42
#ET Time point 6 PC vs AB
```{r Exact time point 6 PC vs AB}
#check open packages
(.packages())
##Close all
detachAllPackages <- function() {
  
  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")
  
  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]
  
  package.list <- setdiff(package.list,basic.packages)
  
  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE)
  
}

detachAllPackages()


library(phyloseq)  
library(ggplot2)
library(microbiome)
library(dplyr)
library(plyr)
library(scales)
library("MultNonParam")
library('PMCMR')


#Convert phyloseq OTU count data into DGEList for edgeR package
##http://joey711.github.io/phyloseq-extensions/edgeR.html

#to install edgeR
#source("https://bioconductor.org/biocLite.R")
#biocLite("edgeR")

phyloseq_to_edgeR = function(physeq, group, method = "RLE", ...) {
  require("edgeR")
  require("limma")
  require("phyloseq")
  # Enforce orientation.
  if (!taxa_are_rows(physeq)) {
    physeq <- t(physeq)
  }
  x = as(otu_table(physeq), "matrix")
  # Add one to protect against overflow, log(0) issues.
  x = x + 1
  # Check `group` argument
  if (identical(all.equal(length(group), 1), TRUE) & nsamples(physeq) > 1) {
    # Assume that group was a sample variable name (must be categorical)
    group = get_variable(physeq, group)
  }
  # Define gene annotations (`genes`) as tax_table
  taxonomy = tax_table(physeq, errorIfNULL=FALSE)
  if( !is.null(taxonomy) ){
    taxonomy = data.frame(as(taxonomy, "matrix"))
  } 
  # Now turn into a DGEList
  y = DGEList(counts = x, group = group, genes = taxonomy, remove.zeros = TRUE, 
              ...)
  # Calculate the normalization factors
  z = calcNormFactors(y, method = method)
  # Check for division by zero inside `calcNormFactors`
  if (!all(is.finite(z$samples$norm.factors))) {
    stop("Something wrong with edgeR::calcNormFactors on this data,\n         non-finite $norm.factors, consider changing `method` argument")
  }
  # Estimate dispersions
  return(estimateTagwiseDisp(estimateCommonDisp(z)))
}


dge10 = phyloseq_to_edgeR(Fungi_T6_PC_AB, group = "Treatment")

# Perform binary test
et10 = exactTest(dge10)

# Extract values from test results
tt10 = topTags(et10, n = nrow(dge10$table), adjust.method = "BH", sort.by = "PValue")
res10 = tt10@.Data[[1]]
alpha = 0.05
F_sigtab_T6_PC_AB = res10[(res10$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                                             
F_sigtab_2fold_T6_PC_AB <- subset(F_sigtab_T6_PC_AB, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T6_PC_AB)

#
F_sigtab_2fold_T6_PC_AB <- data.frame(OTU = rownames(F_sigtab_2fold_T6_PC_AB), F_sigtab_2fold_T6_PC_AB)
```


#ET Time point 6 PC vs AF 
```{r Exact time point 6 PC vs AF}

dge11 = phyloseq_to_edgeR(Fungi_T6_PC_AF, group = "Treatment")

# Perform binary test
et11 = exactTest(dge11)
# Extract values from test results
tt11 = topTags(et11, n = nrow(dge11$table), adjust.method = "BH", sort.by = "PValue")
res11 = tt11@.Data[[1]]
alpha = 0.05
F_sigtab_T6_PC_AF = res11[(res11$FDR <= alpha), ]

#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                                                 
F_sigtab_2fold_T6_PC_AF <- subset(F_sigtab_T6_PC_AF, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T6_PC_AF)

#
F_sigtab_2fold_T6_PC_AF <- data.frame(OTU = rownames(F_sigtab_2fold_T6_PC_AF), F_sigtab_2fold_T6_PC_AF)

```


#ET Time point 6 PC vs AB_AF 
```{r Exact time point 6 PC vs AB_AF}

dge12 = phyloseq_to_edgeR(Fungi_T6_PC_AB_AF, group = "Treatment")

# Perform binary test
et12 = exactTest(dge12)
# Extract values from test results
tt12 = topTags(et12, n = nrow(dge12$table), adjust.method = "BH", sort.by = "PValue")
res12 = tt12@.Data[[1]]
alpha = 0.05
F_sigtab_T6_PC_AB_AF = res12[(res12$FDR <= alpha), ]


#subset by most significant only (p value <0.05 and at least a 2-fold change in abudnance)                                                                 
F_sigtab_2fold_T6_PC_AB_AF <- subset(F_sigtab_T6_PC_AB_AF, FDR < 0.05 & logFC >= 2 | logFC <= -2 & FDR < 0.05)
head(F_sigtab_2fold_T6_PC_AB_AF)

#
F_sigtab_2fold_T6_PC_AB_AF <- data.frame(OTU = rownames(F_sigtab_2fold_T6_PC_AB_AF), F_sigtab_2fold_T6_PC_AB_AF)
```

#Large df for ET at time point 6
#Make one list of which the organisms have been shown to change from the Exact test.
#Combine exact test at the genus level 
```{r Make one dataframe for sig OTU's at Genus level }

#make one data frame of ET according to treatment then subset phyloseq to only keep the significant ones at that particular time point
F_FDRcorrected_T6=c(as.character(F_sigtab_2fold_T6_PC_AB[,1]),as.character(F_sigtab_2fold_T6_PC_AF[,1]), as.character(F_sigtab_2fold_T6_PC_AB_AF[,1]))

#Subset the taxa by the OTUs that were shown to change significantly) 
F_Subset_sig_OTU_T6_genus <-subset_taxa(Fungi_phyloseq, OTU %in% F_FDRcorrected_T6) 

#Write csv to edit names in excel but since all names are informative no editting needed
write.csv(tax_table(F_Subset_sig_OTU_T6_genus),file="F_sigOTU_T6.csv")

#Glom at genus level
F_Subset_sig_OTU_T6_genus_df <- tax_glom(F_Subset_sig_OTU_T6_genus, taxrank = 'Genus') %>%#Merge the species at the genus level
psmelt() #Melt it into a dataframe  #Error no taxonomy table

```


#Manually sort data by response towards urea and inhibitors by adding these two columns to data frame in excel and then export back into R
```{r summary significant OTU}

#Calculate summary 
F_All_sig_OTU_Gen_T6_df <- summarySE(F_Subset_sig_OTU_T6_genus_df, measurevar = "Abundance", groupvars=c("Treatment", "Genus","Phylum"))


F_All_sig_OTU_Gen_T6_df<-dplyr::arrange(F_All_sig_OTU_Gen_T6_df, Genus, Abundance) #Arrange the rows by listed variables

#write csv for F_All_sig_OTU_Gen_T6_df and manually add columns for response towards urea and inhibitors
write.csv(F_All_sig_OTU_Gen_T6_df,file="F_All_sig_OTU_Gen_T6_df.csv")

#read file with response columns added to it,delete number column from F_All_sig_OTU_Gen_T6
F_T6_growth_response=read.csv("/Volumes/S_ GANASAMU/Paper 3/Section 4/F_All_sig_OTU_Gen_T6_df_edited.csv")

```


#Subset by response towards urea and inhibitors 
```{r}

F_T6_growth_response<-dplyr::arrange(F_T6_growth_response, Response_to_urea,Response_to_Inhibitors) #Arrange the rows by listed variables


F_T6_growth_response$Treatment<-factor(F_T6_growth_response$Treatment, levels=c("Negative Control","Positive Control", "Antibacterial", "Antifungal","Antibacterial+Antifungal"))

#Subset df by fast, positive
FT6_Positive_Stimulated=subset(F_T6_growth_response,Response_to_urea=="Positive" & Response_to_Inhibitors=="Stimulated")

FT6_Positive_Inhibited=subset(F_T6_growth_response,Response_to_urea=="Positive" & Response_to_Inhibitors=="Inhibited")

FT6_Negative_Inhibited=subset(F_T6_growth_response,Response_to_urea=="Negative" & Response_to_Inhibitors=="Inhibited")

```

#Figure 6
#T6 Plot Positive_Stimulated
```{r}

#save plot using ggsave

#Plot
FT6_Positive_Stimulated_plot=ggplot(FT6_Positive_Stimulated, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Genus))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1,size=15),strip.text = element_text(size=15),axis.text.y = element_text(size=15), axis.title=element_text(size=20,face="bold"),legend.text=element_text(size=20),legend.title=element_text(size=20,face="bold"))+
 scale_fill_manual(values=FungiGrowthrescolourlistT6)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=2,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x axis tick mark labels
FT6_Positive_Stimulated_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

ggsave("/Volumes/S_ GANASAMU/Paper 3/Section 4/5.pdf",width=9,height=3)
```

#Fugure 6
#T6 Plot Positive_Inhibited
```{r}

#save plot using ggsave

#Plot
FT6_Positive_Inhibited_plot=ggplot(FT6_Positive_Inhibited, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Genus))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
 theme(axis.text.x = element_text(angle = 45, hjust = 1,size=15),strip.text = element_text(size=15),axis.text.y = element_text(size=15), axis.title=element_text(size=20,face="bold"),legend.text=element_text(size=20),legend.title=element_text(size=20,face="bold"))+
 scale_fill_manual(values=FungiGrowthrescolourlistT6)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=2,labeller = label_wrap_gen(width = 10, multi_line=TRUE))


#Rename x axis tick mark labels
FT6_Positive_Inhibited_plot +scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

ggsave("/Volumes/S_ GANASAMU/Paper 3/Section 4/6.pdf",width=9,height=3)
```

#Figure 6
#T6 Plot Negative_Inhibited
```{r}

#save plot using ggsave

#Plot
FT6_Negative_Inhibited_plot=ggplot(FT6_Negative_Inhibited, 
                         aes(x=Treatment, 
                             y=Abundance, 
                             fill=Genus))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  theme_bw() + 
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Treatment")+ geom_errorbar(aes(ymin=Abundance-se, ymax=Abundance+se), colour="black", position=pd, width=0.25) +
  ylab("Abundance")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1,size=15),strip.text = element_text(size=15),axis.text.y = element_text(size=15), axis.title=element_text(size=20,face="bold"),legend.text=element_text(size=20),legend.title=element_text(size=20,face="bold"))+
 scale_fill_manual(values=FungiGrowthrescolourlistT6)+
facet_wrap(.~Genus,drop = TRUE,scales = "free_y",ncol=4,labeller = label_wrap_gen(width = 10, multi_line=TRUE))

#Rename x axis tick mark labels
FT6_Negative_Inhibited_plot+scale_x_discrete(labels=c("Negative Control" = "NC", "Positive Control" = "PC",
                              "Antibacterial" = "AB","Antifungal"="AF","Antibacterial+Antifungal"="AB+AF"))

ggsave("/Volumes/S_ GANASAMU/Paper 3/Section 4/7.pdf",width=9,height=3)
```



